litestar/di.py

```diff
@@ -28,12 +29,14 @@ class Provide:
         "has_sync_callable",
         "has_sync_generator_dependency",
         "has_async_generator_dependency",
+        "parsed_signature",
         "signature_model",
         "sync_to_thread",
         "use_cache",
         "value",
     )
 
+    parsed_signature: ParsedSignature
```

Comment:
I've added the parsed sig to the DI interface, as we need to create it to make the signature model, and we've been accessing the parameters from the parsed signature via `signature_model._fields`, when we may as well access them directly from the source.